src/common/numa.h

```diff
@@ -3,6 +3,13 @@
 
 #pragma once
 
+#if defined(__FreeBSD__)
+#include <sys/param.h>
+#include <sys/cpuset.h>
+#define cpu_set_t cpuset_t
+int sched_setaffinity(pid_t pid, size_t cpusetsize,
```

Comment:
why would you want to define this function here? presumably, it should be in `compat.c`. and it should be defined using `cpuset_setaffinity()`.

Reply:
@tchaikov 
When going for the implementation I would implement it here I think?
So that's why it is declared here.

Reply:
@tchaikov 
Could be that `<sys/cpuset.h>` can do this.
It holds `cpuset_{set,get}affinity()`

Reply:
@tchaikov 
Reason to do it here is that @liewegas already created numa.cc to get things into the shape he required. So I'm just adding to that. Adding a FreeBSD version of `sched_setaffinity()` here, keeps this all in once location.

Reply:
i think the same applies to quite a few stuff we added to `compat.h` and `compat.cc`. see also the discussion at https://github.com/ceph/ceph/pull/25250#discussion_r236174848

Reply:
@tchaikov 
Yes, I think that that will be a question every time FreeBSD and Linux are not in sync the typical OS type things.... For the code I added in `compat.cc` I first had a separate file, and you persuaded me to create `compat.cc`. So lets se when we get to the implementation. This is mainly to get `master` compiling again.